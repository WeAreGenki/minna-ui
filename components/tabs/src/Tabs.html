<!--
  TABS

  NOTE: This component is currently an experimental work in progress and may not
  work correctly. It should not be used in production.
-->

<div class="tabs">
  <nav>
    <a href="#1" class="tabs-title{active === 1 ? ' tabs-title-active' : ''}" on:click|preventDefault="handleClick(event, 1)">1</a>
    <a href="#2" class="tabs-title{active === 2 ? ' tabs-title-active' : ''}" on:click|preventDefault="handleClick(event, 2)">2</a>
    <a href="#3" class="tabs-title{active === 3 ? ' tabs-title-active' : ''}" on:click|preventDefault="handleClick(event, 3)">3</a>
    <a href="#4" class="tabs-title{active === 4 ? ' tabs-title-active' : ''}" on:click|preventDefault="handleClick(event, 4)">4</a>
    <a href="#5" class="tabs-title{active === 5 ? ' tabs-title-active' : ''}" on:click|preventDefault="handleClick(event, 5)">5</a>
    <a href="#6" class="tabs-title{active === 6 ? ' tabs-title-active' : ''}" on:click|preventDefault="handleClick(event, 6)">6</a>
  </nav>
  <div class="tabs-content">
    {#if active === 1}<slot name="1"></slot>{/if}
    {#if active === 2}<slot name="2"></slot>{/if}
    {#if active === 3}<slot name="3"></slot>{/if}
    {#if active === 4}<slot name="4"></slot>{/if}
    {#if active === 5}<slot name="5"></slot>{/if}
    {#if active === 6}<slot name="6"></slot>{/if}
  </div>
</div>

<script>
  export default {
    data: () => ({
      active: 1,
    }),
    oncreate() {
      // console.log('@@ THIS', this);
    },
    methods: {
      handleClick(event, index) {
        // console.log('@@ EVENT', event, index);
        this.set({ active: index });
      },
    },
  };
</script>

<style type="text/postcss">
  @import './_variables.css';
  @import 'import.css';

  .tabs-title {
    padding: 1em;
    border-bottom: 3px solid transparent;

    &:hover,
    &:focus {
      text-decoration: none;
      border-color: $gray5;
    }
  }

  .tabs-title-active {
    border-color: $intent-primary;
  }
</style>
