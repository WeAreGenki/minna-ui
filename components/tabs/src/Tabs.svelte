<!--
  TABS

  @experimental Work in progress. Does not work. DO NOT USE IN PRODUCTION.
-->

<!--
  FIXME: This should be way easier once svelte rfc pr#12 lands:
  https://github.com/sveltejs/rfcs/blob/better-composition/text/0000-better-composition.md
-->

<!--
<script>
  // reactive data
  let active = 1;

  // FIXME: Prob don't need event
  function handleClick(event, index) {
    active = index;
  }
</script>
-->

<style type="text/postcss">
  @import './_tabs.css';
  @import 'import.css';

  /* .tabs-title {
    padding: 1em;
    border-bottom: 3px solid transparent;

    &:hover,
    &:focus {
      text-decoration: none;
      border-color: $gray5;
    }
  }

  .tabs-title-active {
    border-color: $intent-primary;
  } */
</style>

<svelte:options tag="minna-tabs" />

<div class="tabs">
  <!--
  <nav>
    <a href="#1" class="tabs-title {active === 1 ? 'tabs-title-active' : ''}" on:click|preventDefault="{event => handleClick(event, 1)}">1</a>
    <a href="#2" class="tabs-title {active === 2 ? 'tabs-title-active' : ''}" on:click|preventDefault="{event => handleClick(event, 2)}">2</a>
    <a href="#3" class="tabs-title {active === 3 ? 'tabs-title-active' : ''}" on:click|preventDefault="{event => handleClick(event, 3)}">3</a>
  </nav>
  <div class="tabs-content">
    {#if active === 1}<slot name="1"></slot>{/if}
    {#if active === 2}<slot name="2"></slot>{/if}
    {#if active === 3}<slot name="3"></slot>{/if}
  </div>
  -->
</div>
