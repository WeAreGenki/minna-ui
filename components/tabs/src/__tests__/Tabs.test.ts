import Tabs from '../Tabs.svelte';

describe('Tabs component', () => {
  it('renders correctly with defaults', () => {
    const target = document.createElement('div');
    new Tabs({ target });
    expect(target.innerHTML).toMatchSnapshot();
  });
});
