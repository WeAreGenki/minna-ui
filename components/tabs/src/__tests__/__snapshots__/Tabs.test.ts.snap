// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component renders correctly with defaults 1`] = `"<div class=\\"tabs\\"></div>"`;
